vpn/dylib/lib.go

```diff
@@ -0,0 +1,91 @@
+//go:build darwin
+
+package main
+
+import "C"
+
+import (
+	"context"
+	"io"
+	"os"
+
+	"golang.org/x/sys/unix"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/sloghuman"
+
+	"github.com/coder/coder/v2/vpn"
+)
+
+// OpenTunnel creates a new VPN tunnel by `dup`ing the provided 'PIPE'
+// file descriptors for reading, writing, and logging.
+//
+//export OpenTunnel
+func OpenTunnel(cReadFD, cWriteFD int32) int32 {
+	ctx := context.Background()
+
+	conn, err := newFdConn(cReadFD, cWriteFD)
+	if err != nil {
+		return -1
+	}
+
+	// Logs will be sent over the protocol
+	_, err = vpn.NewTunnel(ctx, slog.Make(sloghuman.Sink(io.Discard)), conn)
```

Comment:
There is no point in formatting logs for humans just to discard them. A Logger without sinks is still valid.